lxd/main_forkproxy.go

```diff
@@ -1006,13 +1004,11 @@ func getListenerFile(protocol string, addr string) (*os.File, error) {
 	}
 
 	file := &os.File{}
-	switch listener.(type) {
+	switch l := listener.(type) {
 	case *net.TCPListener:
-		tcpListener := listener.(*net.TCPListener)
```

Comment:
This could be replaced with just `fallthrough` right?

Reply:
I'm not sure I follow. If the type of `listener` is `*net.TCPListener` why would we fall through to where the type is `*net.UnixListener`?

Reply:
Because we do the same thing for both types of listener, which is to assign `file, err = l.File()`, so we what we are really saying is:

"check listener type is unix or tcp, and if not error", then "assign l.File() to file"

Reply:
There's also a bug in there apparently, what happens if the listener isn't either type?

Reply:
Infact I don't really know why we dont just return `l.File()`?
Whats the switch statement for anyway?

Reply:
Right I see so its effectively casting `l` to different types, so we do need to have each case there, otherwise `fallthrough` will end up at the default case (if we had one).

Reply:
I've added a default case returning an error if the listener type is invalid.